Bottom: bd31454b79a6269289ee974e5a3f2ad11be793c1
Top:    3a913e2d815b0b9c94166832305c2dd5f1c5d6cb
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-30 18:22:33 +0800

Refresh of autofs-5.0.7-add-dumpmap-to-file.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index d53235a..c06f240 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2161,6 +2161,8 @@ int main(int argc, char *argv[])
 	}
 
 	if (dumpmaps) {
+		struct master_mapent *entry;
+		struct list_head *p;
 		struct mapent_cache *nc;
 
 		open_log();
@@ -2177,6 +2179,14 @@ int main(int argc, char *argv[])
 
 		lookup_nss_read_master(master_list, 0);
 		master_show_mounts(master_list, dump);
+
+		p = master_list->next;
+		while (p != master_list) {
+			entry = list_entry(p, struct master_mapent, join);
+			p = p->next;
+			master_free_mapent_sources(entry, 1);
+			master_free_mapent(entry);
+		}
 		master_kill(master_list);
 		if (dump)
 			free((void *) dump);
